Packages/doc/databrowser_paplot.rst

```diff
@@ -0,0 +1,223 @@
+   |Pulse 2, Sweep 1|Pulse 2, Sweep 1|
+   +----------------+----------------+
+   |Pulse 1, Sweep 1|Pulse 2, Sweep 0|
+   +----------------+----------------+
+   |Pulse 0, Sweep 1|Pulse 1, Sweep 1|
+   +----------------+----------------+
+   |Pulse 2, Sweep 0|Pulse 1, Sweep 0|
+   +----------------+----------------+
+   |Pulse 1, Sweep 0|Pulse 0, Sweep 1|
+   +----------------+----------------+
+   |Pulse 0, Sweep 0|Pulse 0, Sweep 0|
+   +----------------+----------------+
+   |Deconvolution   |Deconvolution   |
+   +----------------+----------------+
+   |Average         |Average         |
+   +----------------+----------------+
+
+.. _db_paplot_timealignment:
+
+Time Alignment
+^^^^^^^^^^^^^^
+
+The pulses from different channels might not have the same time basis. Time alignment is a way to correct for these offsets.
+
+The algorithm is as follows:
+
+- Get the feature position ``featurePos`` for all pulses which belong to the
+  same set. Store these feature positions using their sweep number and pulse
+  index as key.
+- Now shift all pulses in all sets from the same region by ``-featurePos``
+  where ``featurePos`` is used from the same sweep and pulse index.
+
+.. _db_paplot_operation_order:
+
+Operation order
+^^^^^^^^^^^^^^^
+
+The operations on the data happens in the following fixed order:
+
+- Gather pulses
+- Apply pulse sort order
+- Failed pulse searching
+- Zeroing
+- Time alignment
+- Averaging
+
+PA plot settings
+^^^^^^^^^^^^^^^^
+
+The following figure shows the browser settings panel for the PA plot. It can
+be reached from the ``PA`` tab in the Databrowser. The next sections explain each control.
+
+.. _Figure Pulse Average Browser Settings:
+
+.. figure:: BrowserSettingsPanel-PA-plot.png
+   :align: center
+
+   Settings for the pulse average plot
+
+Pulse Averaging
+"""""""""""""""
+
+- ``enable``: Toggle the display of the PA image/trace plots.
+  Can be turned off before tweaking the settings and then reenabled to make
+  changing the settings quicker.
+
+Image settings
+""""""""""""""
+
+- ``Enable image plot``: Toggle the display of the :ref:`Figure Pulse Average Image plot`
+- ``Draw X zero line``: Draw a vertical line at ``X == 0`` when
+  time alignment is turned on. Helps with distinguishing true from false responses.
+- ``Popup Menu``: Color scheme for image plots
+- ``Sort order``: Sort order of the shown pulses. For ``Sweep`` the pulses are
+  first ordered by sweep number and then by pulse index. For ``Pulse`` it is
+  first pulse index and then sweep number. See also
+  :ref:`db_paplot_image_sortorder`.
+
+Trace settings
+""""""""""""""
+
+- ``Enable trace plot``: Toggle the display of the :ref:`Figure Pulse Average Trace plot`.
+- ``Vertical scale bar``: Size of the vertical scale bar in y-axis units
+
+Deconvolution
+"""""""""""""
+
+- ``Deconvolution``: Toggle the deconvolution pulse display, this is caculated
+  from the averaged pulses [#1]_. For the image plot, the average is the second
```

Comment:
Enables deconvolution [https://journals.physiology.org/doi/full/10.1152/jn.00942.2007] of the average pulse.

Reply:
Deconvolution trace is displayed with the average trace in the trace plot and immediately above the average line in the image plot.